src/adapters/kafka/index.ts

```diff
@@ -14,16 +15,32 @@ class KafkaAdapter extends Adapter {
     this.kafka = new Kafka({
       clientId: 'glee-app',  // clientID: hardcoded need to change afterwards 
       brokers: [brokerUrl.host],
-      ssl: true,
+      ssl: {
+        rejectUnauthorized: true,
+        ca: [fs.readFileSync('ca.pem')],
+        key: fs.readFileSync('client.key'),
+        cert: fs.readFileSync('client.pem')
+      },
       sasl: {
         mechanism: 'plain',
         username: process.env.KAFKA_USERNAME,
         password: process.env.KAFKA_PASSWORD,
       },
     })
 
-    const consumer = this.kafka.consumer({ groupId: 'glee-group' })   // groupID: hardcoded need to change afterwards
+    const securityRequirements = (this.AsyncAPIServer.security() || []).map(sec => {
+      const secName = Object.keys(sec.json())[0]
+      return this.parsedAsyncAPI.components().securityScheme(secName)
+    })
+    const userAndPasswordSecurityReq = securityRequirements.find(sec => sec.type() === 'userPassword')
+    const X509SecurityReq = securityRequirements.find(sec => sec.type() === 'X509')
+    const url = new URL(this.AsyncAPIServer.url())
 
+    const certsConfig = process.env.GLEE_SERVER_CERTS?.split(',').map(t => t.split(':'))
+    const certs = certsConfig?.filter(tuple => tuple[0] === this.serverName)?.map(t => fs.readFileSync(t[1]))
+    
```

Comment:
I referred to this from the MQTT adapter and was not sure if we should put things in a similar way or how we should put it, so do we have to do this in a similar manner in our adapter as well or everything would get passed in SSL and SASL?
@fmvilas

Reply:
You need to get these values before line 15 because you need to pass them to the SASL config (forget SSL, it has nothing to do with the AsyncAPI file but with the glee.config.js file). I hope that makes sense.